utils/deprecation_Decorators.py

```diff
@@ -0,0 +1,33 @@
+import functools
+import warnings
+
+
+def deprecate_arguments(deprecated_args, deprecated_in, removed_in, current_version):
```

Comment:
Let's add type hints here while we can:

```suggestion
from __future__ import annotations

import functools
import warnings


def deprecate_arguments(deprecated_args: dict[str], deprecated_in: str, removed_in: str, current_version: str):
```